internal/controller/clusterconnect_controller.go

```diff
@@ -558,6 +565,27 @@ func (r *ClusterConnectReconciler) reconcileKubeconfig(ctx context.Context, cc *
 	return nil
 }
 
+func (r *ClusterConnectReconciler) reconcileConnectionProbe(ctx context.Context, cc *v1alpha1.ClusterConnect) error {
+	log.FromContext(ctx)
+	// Initialize ConnectionProbe if not already set.
+	if cc.Status.ConnectionProbe == (v1alpha1.ConnectionProbeState{}) {
+		cc.Status.ConnectionProbe = v1alpha1.ConnectionProbeState{
+			LastProbeTimestamp:        metav1.Time{},
+			LastProbeSuccessTimestamp: metav1.Time{},
+			ConsecutiveFailures:       0,
+		}
+	}
+
+	// TODO: parametrize the threshold for consecutive failures.
+	if cc.Status.ConnectionProbe.ConsecutiveFailures > 5 {
```

Comment:
In my testing, when the connect-agent is down, I see the the ConsecutiveFailures counter increasing by 6 every time (0, 6, 12, 18...).   I'm not sure why and I assume that's unexpected.  As a result the ConnectionProbe is marked as Failed after only 1 minute. 

At any rate, I think a more robust method than counting failed probes would be to look at the difference between time.Now() and the LastProbeSuccessTimestamp. If it's greater than 5 minutes, we can decide the probe has failed.